causalpy/experiments/base.py

```diff
@@ -0,0 +1,63 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""
+Base class for quasi experimental designs.
+"""
+
+from abc import abstractmethod
+
+from causalpy.pymc_models import PyMCModel
+from causalpy.skl_models import ScikitLearnModel
+
+
+class BaseExperiment:
+    """Base class for quasi experimental designs."""
+
+    def __init__(self, model=None, **kwargs):
```

Comment:
What do the kwargs do? They are passed through in the child classes. However, they just seems error prone as people will silently be able to pass additional kwargs

Reply:
Good spot. There were some unnecessary karts being passed in for the DID experiments which were just going into a black hole here. I've removed `**kwargs` from being passed to `BaseExperiment.__init__`